storage/rocksdb/rdb_datadic.h

```diff
@@ -42,6 +42,48 @@ class Rdb_field_packing;
 class Rdb_cf_manager;
 class Rdb_ddl_manager;
 
+class Rdb_merge_operator : public rocksdb::AssociativeMergeOperator {
+ public:
+  virtual bool Merge(const rocksdb::Slice &key,
+                     const rocksdb::Slice *existing_value,
+                     const rocksdb::Slice &value, std::string *new_value,
+                     rocksdb::Logger *logger) const override {
+    
+    DBUG_ASSERT(new_value != nullptr);
+    std::string merged_value;
+
+    /* if no existing vaue, then persist the passed in value */
```

Comment:
Since the merge operator could be extended for other operations, it make sense to determine first which merge operation to perform based on the key. You'll want to check that the key starts with the proper index type first and ignore anything that it doesn't understand, or fail.

We also spoke about adding a versioning scheme to the value to handle extending the format in the future.